### DISCUSSION TITLE
How can docker-slim help in scenarios where the image can not run independently?

### DISCUSSION THREAD

**Discussion by anil181** [slimtoolkit/slim#291] [Q&A] [answered]
Hello docker-slim team, I am exploring this tool to reduce the images' size used in our product. We use Helm chart to deploy the applications in our K8s cluster. The images which we deploy can not be run independently with docker run. 
There are other environment specific configs which are being mounted as config map with Helm Chart and even the exposed ports are being set as ENV during Helm deployment which makes those images run when deployed in K8s.

How can  docker-slim help me to analyse the images in this scenario? Can docker-slim interact with a pod deployed in Kubernets and create a slim image?

Thanks & Regards,
Anil Kumar

**Comment by kcq** [comment, marked as answer]
@anil181 native support for k8s is on the todo list. In the meantime the best option is to create a docker-compose config for your chart. The config maps might need to be converted to config file volume mounts in compose.

**Reply by anil181** [reply]
Thank you @kcq
